### ISSUE TITLE
Custom recipes with circuit as input dont show on JEI

### ISSUE THREAD

**Issue by DevDyna** [GregTechCEu/GregTechCEu-1.19#2629]
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

gtceu-1.21.1-1.4.6

### Minecraft Version

1.21.1 NeoForge

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

No

### Other Installed Mods

Required to reproduce this issue:
JEI

### Expected Behavior

Adding a recipe to a machine that just have one slot and all of default recipes dont include circuits as input it should show info about that or just throw errors on logs

(idk if it was a bug or not)

### Actual Behavior

It work but still "hidden" the circuit on jei due the machine slots / it dont support circuits as input

## step 1:
create a datapack / install kubejs and add inside `kubejs/data/<modname>/recipe/` this recipe

//just the sifter gravel recipe but with stone as input + circuit 1
[new.json](https://github.com/user-attachments/files/18233955/new.json)

## step 2:
run `/reload` and search it on jei

![image](https://github.com/user-attachments/assets/dd035371-5d3c-4a03-a456-31dfd6ace7e2)

## step 3:
place a sifter correctly configured without change circuit settings and it look like that (dont execute the recipe)
![image](https://github.com/user-attachments/assets/3a89eca0-8560-4ca6-93b9-0381a289978d)

## step 4:
change the circuit setting to 1

you can see that it execute the recipe
![image](https://github.com/user-attachments/assets/e8cec55c-de48-41f0-bba8-a12f4bf61cc0)

## Possible solution:
completly change circuit view on jei ( item -> texture ) and adding it a condition to show only when on recipe input still required so it can be used on custom recipes where input slots are just one

**Comment by a-a-GiTHuB-a-a** [comment]
can sifters even accept a circuit?

**Comment by DevDyna** [comment]
> can sifters even accept a circuit?

yes (using a custom recipe using a datapack but probably work at same using kjs )

**Comment by a-a-GiTHuB-a-a** [comment]
It's probably because the circuit uses an item slot which the sifter doesn't have. I don't think it's a GT issue.

**Comment by krossgg** [comment]
While the circuit setting *does* work if you do it this way, it is not meant to. The circuit slot is supposed to count as one of the inputs, and the sifter, and cutter, for instance, do not have this second slot. However, the circuit setting does still work as a side effect of how our machines are defined.

**Comment by a-a-GiTHuB-a-a** [comment]
Should this be closed, then?
